src/main/java/ua/tc/marketplace/controller/CategoryController.java

```diff
@@ -0,0 +1,55 @@
+package ua.tc.marketplace.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ua.tc.marketplace.model.entity.Category;
+import ua.tc.marketplace.service.CategoryService;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/v1/category")
+@RequiredArgsConstructor
+public class CategoryController {
+
+    private final CategoryService categoryService;
+
+    @GetMapping
+    public ResponseEntity<Page<Category>> getAllCategories(Pageable pageable) {
+        Page<Category> categories = categoryService.findAll(pageable);
+        return ResponseEntity.ok(categories);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Category> getCategoryById(@PathVariable Long id) {
+        Optional<Category> category = categoryService.findById(id);
+        return category.map(ResponseEntity::ok)
+                .orElseGet(() -> ResponseEntity.status(HttpStatus.NOT_FOUND).build());
+    }
+
+    @PostMapping
+    public ResponseEntity<Category> createCategory(@RequestBody Category category) {
```

Comment:
you should use dto here. Controller accepts dto, sends it to service. Dto is transformed into entity in Service method. Service make actions with entity (save, update, delete), then it is transformed back to dto and returns back into controller.